.env.example

```diff
@@ -25,6 +25,15 @@ DB_PASSWORD=secret
 # Should CDash automatically remove old builds?
 #AUTOREMOVE_BUILDS=true
 
+# How many builds should CDash try to remove per iteration?
+# Consider tweaking this value if you notice the autoremove queries
+# are slow to execute.
+#AUTOREMOVE_BUILDS_BATCH_SIZE=10
```

Comment:
Instead of adding another configuration option, what do you think about adding some logic to automatically adjust the batch size to keep the running time for a given batch under a set time limit, while maximizing the deletion rate under that time limit.

Reply:
I like the idea of figuring out the batch size automatically rather than making the CDash admin figure it out via trial & error.

Reply:
After thinking about this further, I wonder if we should even bother with batching in the first place.  If we simply delete one at a time, deletion rate will probably be suboptimal, but we are practically guaranteed that the deletion process won't hang.

If each delete is guaranteed to be fast, we can delete for a maximum period of 5 minutes, and schedule the job to run every 10 minutes.  This 5-minutes-on-5-minutes-off approach gets rid of the large monolithic job and keeps cleanup tasks out of the way of submission processing.

Reply:
One-at-a-time is fine with me! Practically speaking, that's the patch I've resorted to applying on large production instances that hang when attempting to delete 100 builds in a single pass.